src/renderers/shared/fiber/ReactFiberReconciler.js

```diff
@@ -134,6 +138,11 @@ module.exports = function<T, P, I, TI, PI, C, CX>(config : HostConfig<T, P, I, T
     },
 
     updateContainer(element : ReactNodeList, container : OpaqueNode, parentComponent : ?ReactComponent<any, any, any>, callback: ?Function) : void {
+      invariant(
+        !ReactFeatureFlags.disableNewFiberFeatures || isValidElement(element),
+        'render(): Invalid component element.'
```

Comment:
I think that's the message for when you return a non-element in a composite component. The one for top-level is here: https://github.com/facebook/react/blob/master/src/renderers/dom/stack/client/ReactMount.js#L439